hnn_core/tests/test_params.py

```diff
@@ -1,15 +1,59 @@
 # Authors: Mainak Jas <mainakjas@gmail.com>
+#          Blake Caldwell <blake_caldwell@brown.edu>
 
 import os.path as op
+import json
+
+import pytest
+from mne.utils import _fetch_file
 
 import hnn_core
-from hnn_core import Params
+from hnn_core import read_params, Params
+hnn_core_root = op.join(op.dirname(hnn_core.__file__), '..')
 
 
 def test_params():
     """Test params object."""
-    hnn_core_root = op.join(op.dirname(hnn_core.__file__), '..')
     params_fname = op.join(hnn_core_root, 'param', 'default.json')
-    params = Params(params_fname)
+    params = read_params(params_fname)
     print(params)
     print(params['L2Pyr*'])
+
+
+def test_legacy_params():
+    """Test reading legacy .param file."""
+    param_url = ('https://raw.githubusercontent.com/hnnsolver/'
+                 'hnn-core/test_data/default.param')
+    params_legacy_fname = op.join(hnn_core_root, 'param', 'default.param')
+    if not op.exists(params_legacy_fname):
+        _fetch_file(param_url, params_legacy_fname)
+
+    params_new_fname = op.join(hnn_core_root, 'param', 'default.json')
+    params_legacy = read_params(params_legacy_fname)
+    params_new = read_params(params_new_fname)
+
+    assert params_new == params_legacy
+
+
+def test_base_params():
+    """Test params object with base params"""
+    param_url = ('https://raw.githubusercontent.com/hnnsolver/'
+                 'hnn-core/test_data/base.json')
+    params_base_fname = op.join(hnn_core_root, 'param', 'base.json')
+    if not op.exists(params_base_fname):
+        _fetch_file(param_url, params_base_fname)
+
+    params_base = read_params(params_base_fname)
+    params = Params()
```

Comment:
I think we should update the docstring of `Params` to indicate that a base set of parameters will be initialized even if nothing is supplied.